src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable } from '@angular/core';
+import { map, switchMap, shareReplay } from 'rxjs/operators';
+import { AppConfig } from '../../../app/app.config';
+import { ApparatusEntriesList, ApparatusEntry, Lemma, Reading, XMLElement } from '../../models/evt-models';
+import { getOuterHTML, xpath } from '../../utils/dom-utils';
+import { arrayToMap } from '../../utils/js-utils';
+import { removeSpaces } from '../../utils/xml-utils';
+import { EditionDataService } from '../edition-data.service';
+import { GenericParserService } from './generic-parser.service';
+import { WitnessesParserService } from './witnesses-parser.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApparatusEntriesParserService {
+  private witnessesNumber$ = this.witParserService.witnesses$.pipe(
+    map((x) => Object.keys(x).length),
+  );
+
+  public readonly appEntries$ = this.witnessesNumber$.pipe(
+    switchMap((n) => this.editionDataService.parsedEditionSource$.pipe(
+      map((source) => this.parseAppEntriesList(source, n)),
+      shareReplay(1),
+    )),
+  );
+
+  private appEntryTagName = 'app';
+  private lemmaTagName = 'lem';
+  private readingTagName = 'rdg';
+  private readingGroupTagName = 'rdgGrp';
+  private noteTagName = 'note';
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+    private witParserService: WitnessesParserService,
+  ) {
+  }
+
+  private parseAppEntriesList(document: XMLElement, witsNumber: number): ApparatusEntriesList {
+    const appEntries = Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName));
+
+    return this.parseAppEntries(appEntries, witsNumber);
+  }
+
+  private parseAppEntries(appEntries: XMLElement[], witsNumber: number) {
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry, witsNumber));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement, witsNumber: number): ApparatusEntry {
+    const entry = {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content: this.parseAppReadings(appEntry),
+      note: this.parseAppNotes(appEntry),
+      variance: 0,
+      originalEncoding: getOuterHTML(appEntry),
+    };
+
+    entry.variance = this.calcVariance(entry.content, witsNumber);
+
+    return entry;
+  }
+
+  private parseAppReadings(appEntry: XMLElement): Array<Lemma | Reading> {
```

Comment:
So lemma is considered a reading?

Reply:
Yes, the lemma is a reading. In the TEI Guidelines:

> the term lemma is used here in the text-critical sense of ‘the reading accepted as that of the original or of the base text’.

Right @ChiaraDipi?